src/core/cells.cpp

```diff
@@ -48,23 +50,24 @@
 CellStructure cell_structure;
 
 /**
- * @brief Get pairs closer than distance from the cells.
+ * @brief Get pairs of particles that are closer than a distance and fulfill a
+ * filter criterion.
  *
- * This is mostly for testing purposes and uses link_cell
- * to get pairs out of the cellsystem by a simple distance
- * criterion.
+ * It uses link_cell to get pairs out of the cellsystem
+ * by a simple distance criterion and
  *
  * Pairs are sorted so that first.id < second.id
  */
-std::vector<std::pair<int, int>> get_pairs(double distance) {
+template <class Filter>
+std::vector<std::pair<int, int>> get_pairs_filtered(double const distance,
+                                                    Filter filter) {
   std::vector<std::pair<int, int>> ret;
+  on_observable_calc();
   auto const cutoff2 = distance * distance;
-
-  cells_update_ghosts(Cells::DATA_PART_POSITION | Cells::DATA_PART_PROPERTIES);
```

Comment:
Are you sure this is not needed anymore?

Reply:
``on_observable_calc()`` does the ghost update

Reply:
Ah right!